# STATISTICS MID EXAM

### Total Marks : 40
### Time - 2 Hours


 _________________

# SECTION A: 10 MARKS

### 1. A sample of 20 students is randomly selected and collected their scores in Data Science.  Perform an interval estimate on population scores with 95% confidence. (5 marks)

datascience_scores= [88, 45, 62, 54, 57, 66, 69, 94, 60, 50, 86, 81, 89, 62, 56, 60, 51,
       94, 87, 55]
       
#### Note : Assume data is normal

In [8]:
import numpy as np 

datascience_scores = np.array([88, 45, 62, 54, 57, 66, 69, 94, 60, 50, 86, 81, 89, 62, 56, 60, 51, 94, 87, 55])

mean = datascience_scores.mean()
std = datascience_scores.std()

mean, std

(68.3, 15.84645070670401)

In [9]:
from scipy.stats import t

n = 20
confidence_level = 0.95
alpha = 1 - confidence_level
dof = n - 1

t_score = t.ppf(confidence_level + alpha / 2, dof)

margin_error = t_score * (std / np.sqrt(n))

t_score, margin_error

(2.093024054408263, 7.4163672213604315)

In [10]:
lower_bound = mean - margin_error
upper_bound = mean + margin_error

lower_bound, upper_bound

(60.883632778639566, 75.71636722136043)

In [11]:
np.round(lower_bound), np.round(upper_bound)

(61.0, 76.0)

### Conclusion

Since we have sample size 20, i.e less than 30, we are using t test.
The 95% confidence interval for the given sample scores on the population lies between **61** to **76**.

### 2. A shopkeeper assumes that equal proportion rice, corn, wheat availble in his store. After measuring , he found that  100 packets of rice, 75 packets of corn, 90 packets of wheat in his shop. Check the farmers assumptions with 99% confidence. ( 5 marks)

In [28]:
# Define the Hypothesis
# H0: All Proportions are equal
# H1: Atleast one Proportion is different


In [29]:
import scipy.stats as stats

observed = np.array([100, 75, 90])
expected = [observed.sum() / 3] * 3

chi2_stats = stats.chisquare(observed, f_exp=expected)[0]

deg_of_freedom = len(observed) - 1
alpha = 0.01

critical_value = stats.chi2.ppf(1 - alpha, deg_of_freedom)

critical_value, chi2_stats

(9.21034037197618, 3.5849056603773586)

In [27]:
if critical_value < chi2_stats:
    print("Reject H0: The proportions are not equal")
else:
    print("Fail to Reject H0: There is not enough evidence to conclude that the proportions are different ")
    

Fail to Reject H0: There is not enough evidence to conclude that the proportions are different 


 _________________

# SECTION B: 10Marks


 _________________

### 3 . A fisherman claims that average weight of fishes in a pond is greater than 3kg. A sample of 50 fishes are selected and found the average weight 3.5 kg and standard devaition of 0.75kg. Test the fisherman's calim with 99% confidence level.

#### Note: Assume data is normal

A. Frame the hypothesis (2 mark)

B. Calculate the test statistic and p_value  (5 mark)

C. Interpret the results (3 marks)

## State the Hypothesis
### H0: The Average weight of the fishes in the pond is lessthan or eqaul to 3 kg.
### H1:The Average weight of the fishes in the pond is geater than 3 Kg

 _________________

In [33]:
# Calculate test statistics
n = 50
x_bar = 3.5
std = 0.75
mu = 3
confidence = 0.99
alpha = 0.01

t_stat = (x_bar - mu) / (std / np.sqrt(n))

t_stat

4.714045207910317

In [32]:
# P value 
p_value = stats.t.sf(t_stat, df=n-1)

p_value

1.0235132411294397e-05

In [34]:
p_value < alpha    

True

## Reject H0:

### Conclusion: The Fishermans claim is True, that Average weight of the fishes in the pond is greater than 3 Kg

# SECTION C: 20 MARKS

 _________________

In [35]:
import pandas as pd
data= pd.read_csv('data_set.csv')
data.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15        9.300          Low Fat         0.016047   
1           DRC01        5.920          Regular         0.019278   
2           FDN15       17.500          Low Fat         0.016760   
3           NCD19        8.930          Low Fat         0.000000   
4           FDP36       10.395          Regular         0.000000   

      Item_Type  Item_MRP Outlet_Identifier  Outlet_Establishment_Year  \
0         Dairy  249.8092            OUT049                       1999   
1   Soft Drinks   48.2692            OUT018                       2009   
2          Meat  141.6180            OUT049                       1999   
3     Household   53.8614            OUT013                       1987   
4  Baking Goods   51.4008            OUT018                       2009   

  Outlet_Size Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  
0      Medium               Tier 1  Supermarket Type1          3735.1380  
1      Medium               Tier 3  Supermarket Type2           443.4228  
2      Medium               Tier 1  Supermarket Type1          2097.2700  
3        High               Tier 3  Supermarket Type1           994.7052  
4      Medium               Tier 3  Supermarket Type2           556.6088

### 4. Frame the hypothesis ,calculate the test statistic and p_value , interpret the results for the following tasks with 95% Confidence level :

i. The shop keeper claims that the average of the item outlet sales is not less than $2280.Validate if his claim is true.(4 marks)

ii. Is there any relation between Item_Fat_Content and Item_Outlet_Sales? (4 marks)

iii.Is there any relation between Outlet_Location_Type and Outlet_Type? (4 marks)

iv.Is there any relation between Item_Weight and Item_Outlet_Sales? (4 marks)

v. Is there any relation between Outlet_Size and Item_Outlet_Sales? (4 marks)


**Note: Assume all continuous columns are normally distributed and have equal variance.**

### 4 - i

#### State Hypothesis
* H0: The Average item outlet sales is less than or eqaul to 2280.
* H1: The Average item outlet sales is greater than 2280.

In [36]:
sample_size = len(data['Item_Outlet_Sales'])
sample_mean = data["Item_Outlet_Sales"].mean()
sample_std = data["Item_Outlet_Sales"].std()

mu = 2280
alpha = 0.05

# Calculaate the t-statistics and p value
t_stat = (sample_mean - mu) / (sample_std / np.sqrt(sample_size))
p_value = t.sf(t_stat, df=sample_size - 1)

t_stat, p_value

(-0.3624724078368445, 0.6414922422772362)

In [37]:
if p_value < alpha:
    print("Reject H0: The Average item outlet sales are greater than $2280")
else:
    print("Fail to reject H0: There is not enough evidence to conclude that the average sales is greater than $2280")

Fail to reject H0: There is not enough evidence to conclude that the average sales is greater than $2280


#### Conclusion: Shopkeeper Claim is not true

#### 4 - iii
H0: No Significant relation between Item_Location_Type and Outlet_Type
H1: There is Relation

In [53]:
from sklearn.preprocessing import LabelEncoder

data["Outlet_Location_Type"] = LabelEncoder().fit_transform(data['Outlet_Location_Type'])
data["Outlet_Type"] = LabelEncoder().fit_transform(data['Outlet_Type'])
# Perform Pearson correlation
correlation_coef = data["Outlet_Location_Type"].corr(data["Outlet_Type"])

# Calculate P-Value
t_stat = correlation_coef * np.sqrt((sample_size - 2) / (1 - correlation_coef ** 2))
p_value = t.sf(t_stat, df=sample_size - 2)

correlation_coef, t_stat, p_value

(0.5582655261035571, 45.87462124439782, 0.0)

#### Conclusion
Since we have got the p_value as **0** i.e less than alpha and  reject H0.
**There is sifnificant relation between Outlet_Location_Type and Outlet_Type**

#### 4 - ii
H0: No Significant relation between Item_Fat_Content and Item_Outlet_Sales
H1: There is Relation

In [50]:
from sklearn.preprocessing import LabelEncoder

data["Item_Fat_Content"] = LabelEncoder().fit_transform(data['Item_Fat_Content'])

# Perform Pearson correlation
correlation_coef = data["Item_Fat_Content"].corr(data["Item_Outlet_Sales"])

# Calculate P-Value
t_stat = correlation_coef * np.sqrt((sample_size - 2) / (1 - correlation_coef ** 2))
p_value = t.sf(t_stat, df=sample_size - 2)

correlation_coef, t_stat, p_value

(0.008847280875777476, 0.603197971700419, 0.27320321427374517)

#### Conclusion
Since we have got the p_value as **0.2732** i.e greter than alpha and failed to reject H0.
**There is no sifnificant relation between Item_Fat_Content and Outlet_Sales**

#### 4 - v
H0: No Significant relation between Outlet_Size and Item_Outlet_Sales
H1: There is Relation

In [54]:
from sklearn.preprocessing import LabelEncoder

data["Outlet_Size"] = LabelEncoder().fit_transform(data['Outlet_Size'])

# Perform Pearson correlation
correlation_coef = data["Outlet_Size"].corr(data["Item_Outlet_Sales"])

# Calculate P-Value
t_stat = correlation_coef * np.sqrt((sample_size - 2) / (1 - correlation_coef ** 2))
p_value = t.sf(t_stat, df=sample_size - 2)

correlation_coef, t_stat, p_value

(0.025965428797625232, 1.770822410412953, 0.038327882612726114)

#### Conclusion
Since we have got the p_value as **0.03** i.e less than alpha, reject H0.
**There is sifnificant relation between Outlet_Size and Outlet_Sales**

#### 4 - iv
H0: No Significant relation between Item_Weight and Item_Outlet_Sales
H1: There is Relation

In [40]:
# Perform Pearson correlation
correlation_coef = data["Item_Weight"].corr(data["Item_Outlet_Sales"])

# Calculate P-Value
t_stat = correlation_coef * np.sqrt((sample_size - 2) / (1 - correlation_coef ** 2))
p_value = t.sf(t_stat, df=sample_size - 2)

correlation_coef, t_stat, p_value

(0.010971826651563646, 0.7480629387621539, 0.2272300210664357)

#### Conclusion
Since we have got the p_value as **0.2272** i.e greter than alpha and failed to reject H0.
**There is no sifnificant relation between Item_Weight and Outlet_Sales**